### PR TITLE
feat: editable scenes instead of snapshots

### PR DESCRIPTION
**Pull request by grant-davidson**
kQY3TaJXaF

### CODE REVIEW

**Comment by dai-shi** [comment]
It doesn't seem to work for moving shapes.


https://user-images.githubusercontent.com/490574/116529181-9f7c5680-a917-11eb-9d64-0736325454b8.mov

**Comment by grant-davidson** [comment]
> It doesn't seem to work for moving shapes.
> 
>  Screen.Recording.2021-04-29.at.18.20.06.mov

That's odd, I'll have a look

**Comment by dai-shi** [comment]
This would be very nice! Did you notice my comment? https://github.com/dai-shi/excalidraw-claymate/issues/9#issuecomment-829059148 It would be nice if we can replace it.

Implementation-wise, I wonder if we need to keep canvas. But, we can revisit it later anyway.

**Comment by grant-davidson** [comment]
> This would be very nice! Did you notice my comment? [#9 (comment)](https://github.com/dai-shi/excalidraw-claymate/issues/9#issuecomment-829059148) It would be nice if we can replace it.
> 
> Implementation-wise, I wonder if we need to keep canvas. But, we can revisit it later anyway.

No, I hadn't seen that but I'll check

**Comment by dai-shi** [comment]
So, right now it's "snapshot", but what would be interesting is "scene"
- we edit one of scene at a time.
- we can change a scene to edit by clicking it (right now "R" button, but can be the entire snapshot area)
- we can "copy scene" and edit it.

It would be easier if you try https://excalidraw-slides.netlify.app/ and see how it behaves.

**Comment by grant-davidson** [comment]
Yes, I think I understand your plan.
You'd be live editing the current snapshot/scene.
When you clicked on other snapshot/scene you'd switch to editing that and the "Add Snapshot" would change it behaviour to copying the current snapshot/scene to a new one which would then become the current scene.

**Comment by dai-shi** [comment]
Exactly!

**Comment by grant-davidson** [comment]
Something like this?
lBoFnUzwsx

**Comment by dai-shi** [comment]
Wow, you are already work on it!
Let me know your plan for this PR.
Is this going to be for what I named "scene" feature?

**Comment by grant-davidson** [comment]
Yes, that's what I was thinking.  It's working but I'm not sure how much is required to make this robust

**Comment by dai-shi** [comment]
Great. I'm not sure either...
One minor request, if you agree, is to change the button to "Copy scene".
We also want to highlight the scene which we are editing.
Unless it won't break the basic use case, we can tackle remaining issues separately.

**Comment by grant-davidson** [comment]
qMyFKOplF8

**Comment by grant-davidson** [comment]
mRFEZd7UIX

**Comment by grant-davidson** [comment]
It's probably worth changing all "snapshot" variables to "scene"

**Comment by dai-shi** [comment]
It works great! I like it.

File: README.md [hunk 1]
```diff
@@ -15,10 +15,11 @@ If you are already familiar with Excalidraw,
 just visit <https://dai-shi.github.io/excalidraw-claymate>
 and you can do the same.
 Only the difference is the bottom area, where you can
-add snapshots and export them as animated GIF.
+add scenes and export them as animated GIF.
 
 Notes:
-- The snapshot size is limited to the size of the first snapshot.
+
+- The scene size is limited to the size of the first scene.
```
Comment (dai-shi) [inline README.md:22]: We want to eventually resolve this limitation. Maybe after we figure out exporting in multiple formats.

File: src/Claymate.css [hunk 2]
```diff
@@ -126,3 +132,13 @@
   height: 24px;
   padding: 0;
 }
+
+.Claymate-restore {
```
Comment (dai-shi) [inline src/Claymate.css:136]: We no longer use this, right?

Comment (grant-davidson) [reply src/Claymate.css:136]: Oh yes, I'll remove it

File: src/types.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export type Drawing = { elements: unknown[]; appState: unknown };
```
Comment (dai-shi) [inline src/types.ts:1]: Maybe we can graph types from excalidraw package.

**Review by dai-shi — APPROVED** [review]
The code looks good too. Great work.

**Pull request merged**